packages/ra-ui-materialui/src/button/Button.tsx

```diff
@@ -114,13 +110,16 @@ interface Props {
     size?: 'small' | 'medium' | 'large';
     redirect?: RedirectionSideEffect;
     variant?: string;
-    // May be injected by Toolbar
-    record?: RaRecord;
+    // May be provided manually
+    record?: RecordType;
```

Comment:
In fact, I fail to see what `record` does here. It's not even used in the code. This somehow seems to forbid usage of the generic `Button` outside of a `record` context, e.g. in a List, a dialog, a sidebar, or a custom page. I wonder why, as `Button` is purely a UI component.